### DISCUSSION TITLE
keine Verbindung nach Einsatz Service Techniker von SENEC

### DISCUSSION THREAD

**Discussion by Zeppelin-PV** [solectrus/solectrus#2877] [Q&A]
Bin gerade am zweifeln
Heute war schon ein Service Techniker von SENEC da und hat die wohl defekte NPU ausgetauscht, sowei ein defektes Batteriemodul ausgebaut.
Nun läuft der Speicher wieder mit 7,5kW und COM 70
aber ich bekomme keine Verbindung mehr zum Speicher
IP Adresse ist die gleiche, nur die MCU Firmware ist jetzt 827

weiß da jemand was ???

**Comment by Zeppelin-PV** [comment]
das gibt es nicht ...
... die getauschte NPU arbeitet mit dem http: protokoll nicht mit dem https:

wo war das nochmal umzustellen in Solectrus ?

**Reply by ledermann** [reply]
Kurios, davon habe ich noch nichts gehört. 

In der `.env`:
```
SENEC_SCHEMA=http
```

Danach `docker compose up -d`

**Comment by ledermann** [comment]
Bist du sicher, dass die IP die gleiche ist? Würde mich etwas wundern, da die neue NPU vermutlich eine andere MAC-Adresse hat (das ist eine Art herstellerübergreifende Seriennummer) und vom Router daher als neues Gerät erkannt werden dürfte. Relevant sollte die IP-Adresse sein, die im Display des Speichers steht. 

Woher weißt du denn, dass eine neue Firmware drauf ist, wenn du dich gar nicht mit dem Speicher verbinden kannst? Oder meinst du, dass du mit dem Browser auf die Web-Oberfläche des Speichers kommt, nur SOLECTRUS hingegen nicht funktioniert?

Also:
- Kläre erst, ob du mit dem Browser per https://192.... auf den Speicher kommst
- Wenn ja, aber SOLECTRUS keine Verbindung zeigt: Dann hat der SENEC-Collector Probleme. Prüfe dessen Logs (`docker compose logs senec-collector`). Finden sich da Fehlermeldungen?
- Steht in der `.env` die korrekte IP-Adresse? Ggfs. korrigieren.
- Abschließend `docker compose down && docker compose up -d`

Wenn das nicht hilft, poste bitte die Logs des SENEC-Collectors.

**Comment by Zeppelin-PV** [comment]
ich komme definitiv mit http auf den Senec
mit https nicht mehr
die neu gezogene IP xxx.xxx.xxx.070 habe ich sofort wieder in der FritzBox geändert auf ...230 so dass alles stimmt
die MCU 827 hat der Techniker angestossen, wurde nicht geholt da auf Fernabschaltung

senec-collector-1  | Getting state names (language: de) from SENEC by parsing source code...
senec-collector-1  | Error getting data from SENEC at https://192.168.001.230: connection refused: 192.168.001.230:80

**Comment by Zeppelin-PV** [comment]
```
# SENEC Collector
#
# Change this to your own SENEC IP address!
SENEC_HOST=192.168.001.230
SENEC_INTERVAL=5
SENEC_SCHEMA=https
```

... muss da wieder http stehen ?

**Reply by ledermann** [reply]
Ja, hatte ich oben bereits geschrieben. `SENEC_SCHEMA` muss entweder `http` oder `https` enthalten. Je nachdem, auf welchem Protokoll dein Speicher erreichbar ist.

Nach Änderung an der `.env` nicht vergessen, ein `docker compose up -d` durchzuführen, damit das wirksam wird.

Ich selbst habe noch die 0826. Wenn sich das ändert, gucke ich mir das hier genauer an.

**Comment by Zeppelin-PV** [comment]
das ist alles verwirrend
ich habe den Techniker per Mail über diese Ungereimtheit informiert
dieser  hat im Moment angerufen und dies bestätigt, er schubst nochmal ein Update auf den Senec so dass es wieder https: ist
trotzdem die kurzfristige Umstellung auf http hat funktioniert. Solectrus läuft
wenn dann das Update kommt, alles zurück auf Los, so wie bei Monopoly  ;-)

**Reply by ledermann** [reply]
Ich könnte mir vorstellen, dass auf dem ausgetauschten NPU eine alte Softwareversion drauf ist. Diese arbeitet noch mit `http`. Erst ein Software-Update sorgt für `https`. Beachte, dass die verschiedenen Software-Kompnenten verschiedene Versionsnummer haben.

Achte mal auf Anzeige von "Revision NPU-Image". Bei der Umstellung auf `https` [damals im Herbst 2023](https://solectrus.de/blog/2023-09-06-erfahrungen-mit-dem-senec-update/) war es so, dass nicht das Firmware-Update (MCU) dafür verantwortlich war, sondern erst das NPU-Image.

**Comment by Zeppelin-PV** [comment]
ja, Du hast recht. 
manchmal sieht man den Wald vor lauten Bäumen nicht
Der Techniker hat mir da wohl ein uralt Teil eingebaut
... nur so mal für mich zur Erinnerung: kann ich diesen verkorksten Tag von Senec holen als csv, oder geht da nur eine Woche ?

![grafik](https://github.com/solectrus/solectrus/assets/126392805/b5b3b356-00a5-4bde-a96b-7b04f03731e2)

![grafik](https://github.com/solectrus/solectrus/assets/126392805/218e3f35-fdf9-4270-b7d1-02183144628d)

**Reply by ledermann** [reply]
CSV-Datei der laufenden Woche runterladen bei mein-senec.de (die wird dort laufend aktualisiert), dann Importieren über den CSV-Importer. Das sollte die Lücke füllen.

**Comment by Zeppelin-PV** [comment]
meine zwei Raspi laufen derzeit wieder mit dem http Protokoll
die .env wurde umgestellt
nun kann ich aber keinen Download der LOG daten mehr machen

admin@raspberrypi:~ $ sudo ./download.sh
Downloading from https://192.168.001.230//Log/2024/02/28.log
curl: (7) Failed to connect to 192.168.001.230 port 443: Verbindungsaufbau abgelehnt

Die download.sh ruft demnach mit https die Daten auf, was ja momentan nicht geht

gibt es hierzu eine Hilfe ?

**Comment by Zeppelin-PV** [comment]
... in der download.sh steht:

```
# Define base URL
base_url="https://$senec_ip//Log"
```

könnte man dies nicht auch von der .env holen ?

**Reply by ledermann** [reply]
Die eine Zeile in der `download.sh` kannst du eigentlich leicht selbst ändern. 

Ich habe dennoch jetzt den Log-Downloader so abgeändert, dass die Konfiguration nicht mehr in der `download.sh` erfolgt, sondern in einer `.env`, die im gleichen Ordner liegen muss. Dadurch kann eine bereits existierende .env (von SOLECTRUS) benutzt werden.

Wenn du das nutzen möchtest, kannst du den Log-Downloader hiermit aktualisieren:

```shell
curl -sSO https://raw.githubusercontent.com/solectrus/senec-log-downloader/main/download.sh
``` 

Aber achte unbedingt darauf, dass die `.env` im gleichen Ordner liegt wie `download.sh`.

Bei Problemen mit dem Log-Downloader bitte im dortigen Repo ein Issue anlegen:
https://github.com/solectrus/senec-log-downloader

**Comment by hambue** [comment]
Hat sich denn durch den Austausch des NPU etwas verbessert oder kommt es weiterhin zu "Communication errors"?

**Reply by Zeppelin-PV** [reply]
... ja, seither keine NPU Ausfälle mehr
aber,
seit dem Austausch läuft das System wieder im http Protokoll, und nicht mit https
der Techniker meinte, dass nach dem Austausch wohl einige Tage vergehen bis sich das System das Update auf https holt,
das war am 1. März, aber bis heute ist nichts passiert, vielleicht besser so, wer weiß ....

**Reply by hambue** [reply]
Ok, danke für die Antwort. Solange alles noch auf http läuft, sind wohl eh keine Fehler zu erwarten. Die kamen ja erst im September mit dem Update auf https.

**Comment by MartinKBroll** [comment]
Bei mir ist jetzt dasselbe passiert. Austausch der NPU wegen zweier Selbstabschaltungen des SENEC V3 durch NPU-Fehler. Auf der ausgetauschten NPU ist eine alte Firmware und NPU-Image. Anruf beim Support. Die Mitarbeiterin war erst mal überfordert und nach meinem Insistieren und ihrem Befragen zweier KollegÏnnen war dann klar, dass in Kürze die neue Firmware kommt - vom Image hat sie jetzt nicht gesprochen. Ich habe die .env Datei geändert und den Container neu composed, SOLECTRUS läuft wieder. Jetzt muss ich mir noch die fehlenden Daten importieren.

Es ist natürlich typisch SENEC, dass man über die (temporären) Konsequenzen des NPU Tauschs nicht aufgeklärt wird.

**Reply by MartinKBroll** [reply]
Update: Kurz nachdem ich den Beitrag verfasst hatte war das NPU-Image schon erneuert auf 2310. Die Auskunft, dass es ein paar Tage dauert, war dann falsch.
